code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -108,6 +122,56 @@
 	. += emissive_appearance(icon, "locked", alpha = src.alpha)
 	. += locked ? "locked" : "unlocked"
 
+/// Animates the closet door opening and closing
+/obj/structure/closet/proc/animate_door(closing = FALSE)
+	if(!door_anim_time)
+		return
+	if(!door_obj)
+		door_obj = new
+	vis_contents += door_obj
+	door_obj.icon = icon
+	door_obj.icon_state = "[icon_door || icon_state]_door"
+	is_animating_door = TRUE
+	var/num_steps = door_anim_time / world.tick_lag
+
+	for(var/step in 0 to num_steps)
+		var/angle = door_anim_angle * (closing ? 1 - (step/num_steps) : (step/num_steps))
+		var/matrix/door_transform = get_door_transform(angle)
+		var/door_state
+		var/door_layer
+
+		if (angle >= 90) {
+			door_state = "[icon_door_override ? icon_door : icon_state]_back"
+			door_layer = FLOAT_LAYER
+		} else {
+			door_state = "[icon_door || icon_state]_door"
+			door_layer = ABOVE_MOB_LAYER
+		}
```

Comment:
I love you tattle, but the {} is against how we write procs.
Just going by indentation will work fine

Reply:
My brain is on autopilot again 😔

Reply:
the c likes are rotting your mind

Reply:
when will this codebase get Bracketpilled